assets/js/googlesitekit/data/utils.test.js

```diff
@@ -124,6 +128,231 @@ describe( 'data utils', () => {
+					return null;
+				},
+			},
+		};
+
+		const storeSeven = {
+			resolvers: {
+				*getFakeActionOne() {
+					yield combineStoresFakeActionOne.fakeActionOne();
+				},
+			},
+		};
+
+		const storeEight = {
+			selectors: {
+				getOne: ( state ) => {
+					return state.one;
+				},
+			},
+		};
+
+		it( 'should combine multiple stores into one', () => {
+			const expectedCombinedStore = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo ) ) ).toBe( JSON.stringify( expectedCombinedStore ) );
+		} );
+
+		it( 'should modify combined state', () => {
+			const combinedStore = combineStores( storeOne, storeTwo );
+			let state = combinedStore.reducer();
+
+			expect( state ).toEqual( { one: 1, two: 2 } );
+
+			// It should still respond to the original actions.
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionOne() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionTwo() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+		} );
+
+		it( 'should combine stores missing any of the keys', () => {
+			const combinedStoreMissing = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE, storeThree.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo, storeThree ) ) ).toBe( JSON.stringify( combinedStoreMissing ) );
+
+			const combinedStoreMissingTwo = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer, storeFour.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo, storeFour ) ) ).toBe( JSON.stringify( combinedStoreMissingTwo ) );
+		} );
+
+		it( 'INITIAL_STATEs, reducers, and actions should all work together even if provided by separate incomplete stores', () => {
```

Comment:
This test case is better but there aren't any tests that completely omit a key for every store. I think it'd be worth adding a test with stores that completely lack every key:

* `INITIAL_STATE`
* `actions`
* `controls`
* `reducer`
* `resolvers`
* `selectors`

Reply:
👍